client/components/site-selector/index.jsx

```diff
@@ -222,38 +225,48 @@ export default React.createClass( {
 	},
 
 	renderRecentSites() {
-		const sites = this.props.sites.getRecentlySelected();
-
-		if ( ! sites || this.state.search || ! this.shouldShowGroups() ) {
-			return null;
+		if ( this.state.search || ! this.shouldShowGroups() ) {
+			return;
 		}
 
-		const recentSites = sites.map( function( site ) {
-			var siteHref;
-
-			if ( this.props.siteBasePath ) {
-				siteHref = this.getSiteBasePath( site ) + '/' + site.slug;
-			}
-
-			const isSelected = this.isSelected( site );
-
-			return (
-				<Site
-					site={ site }
-					href={ siteHref }
-					key={ 'site-' + site.ID }
-					indicator={ this.props.indicator }
-					onSelect={ this.onSiteSelect.bind( this, site.slug ) }
-					isSelected={ isSelected }
-				/>
-			);
-		}, this );
+		const { recentSites } = this.props;
+		if ( ! size( recentSites ) ) {
+			return;
+		}
 
-		if ( ! recentSites ) {
-			return null;
+		const sites = this.props.sites.get();
+		if ( ! sites ) {
+			return;
 		}
 
-		return <div className="site-selector__recent">{ recentSites }</div>;
+		return (
+			<div className="site-selector__recent">
+				{ reduce( sites, ( memo, site ) => {
```

Comment:
Let's remember to update this later when we have a bit more ported. We should be able to avoid reducing over all sites.